app/[[...path]]/layout.tsx

```diff
@@ -0,0 +1,20 @@
+import type { ReactNode } from 'react';
+import { createElement } from 'react';
+
+import { Layout as RootLayout } from '@/theme/_layout';
+import router from '@/theme/_router';
+
+export default async function Layout(props: { children: ReactNode; params: { path?: string[] } }) {
```

Comment:
I believe this can become a static layout (without a way to override it on the page module level). This way the out-of-the-box Next.js optimizations should still work for layouts.

Reply:
According to Next docs, the root `layout` component is required anyway. However, it is only intended to provide the boilerplate HTML, since that will not be re-rendered from route to route.

See https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required

For layouts that should be re-rendered on each request, there are [Template](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#templates) (although not sure if the support RSC)

Reply:
FWIW, this _is_ the root layout, as the root catch-all route is viewed by Next.js as the root location.